packages/design-system/src/components/ButtonGroup/ButtonGroup.js

```diff
@@ -8,43 +8,43 @@ const baseClass = 'btn-group';
 
 class ButtonGroup extends React.Component {
   state = {
-    currentIndex: defaultIndex
+    index: defaultIndex
   };
 
   static getDerivedStateFromProps(props, state) {
     return {
-      currentIndex:
+      index:
         typeof props.currentIndex === 'number'
           ? props.currentIndex
           : state.currentIndex
     };
   }
 
   handleClick = (index, event) => {
-    this.props.onIndexChange(index, event);
+    this.props.onIndexChange && this.props.onIndexChange(index, event);
     this.setState({ currentIndex: index });
   };
 
   render() {
-    const { currentIndex } = this.state;
-    const { size, fullWidth, children, className, ...restProps } = this.props;
+    const { index } = this.state;
+    const { size, fullWidth, children, className, onIndexChange, currentIndex, ...restProps } = this.props;
```

Comment:
you do not need `currentIndex` here as it is not used in the component `render` method

Reply:
it's not used, but there is `restProps` from the spread operator. Below div will receive currentIndex as property, but there is no property like this on div element.

Reply:
☝️